WeatherApp/ViewControllers/WeatherDetailViewController.swift

```diff
@@ -10,33 +10,121 @@ import UIKit
+        hourlyWeatherCollectionView.dataSource = self
     }
     
-    required init?(coder: NSCoder) {
-        fatalError("init(coder:) has not been implemented")
+    private func configureCollectionLayout() {
+        let layout = UICollectionViewFlowLayout()
+        layout.itemSize = CGSize(width: 70, height: 150)
+        layout.scrollDirection = .horizontal
+        hourlyWeatherCollectionView.collectionViewLayout = layout
+    }
+    
+    private func setupHourlyWeatherData() {
+        weatherDetailViewModel.getHourlyWeather(completion: { (apiResponseMessage) in
+            switch apiResponseMessage {
+                case .SUCCESSFUL(_): self.updateCollectionView()
+                case .FAILED(let error): print(error)
+            }
+        })
+    }
+    
+    private func setupDailyWeatherData() {
+        weatherDetailViewModel.getDailyWeather { (apiResponseMessage) in
+            switch apiResponseMessage {
+                case .SUCCESSFUL(_): self.updateHourlyStackView()
+                case .FAILED(let error): print(error)
+            }
+        }
     }
     
     private func setupUI() {
-        weatherDetailView.translatesAutoresizingMaskIntoConstraints = false
-        view.addSubview(weatherDetailView)
+        cityLabel.text = weatherDetailViewModel.cityWeather.city
+        dateLabel.text = weatherDetailViewModel.date
+        timeLabel.text = weatherDetailViewModel.time
+        hourlyWeatherCollectionView.backgroundColor = .systemBlue
+    }
+    
+    private func updateCollectionView() {
+        DispatchQueue.main.async {
+            self.hourlyWeatherCollectionView.reloadData()
+        }
+    }
+    
+    private func updateHourlyStackView() {
+        for (index, dailyViews) in self.dailyWeatherViews.enumerated() {
+            guard weatherDetailViewModel.checkDailyForecastCount(with: index) else { return }
+            
+            guard let dayData = self.weatherDetailViewModel.dailyWeather?.dailyForecast[index] else { return }
+            
+            DispatchQueue.main.async {
+                dailyViews.setupView(with: dayData)
+            }
+        }
+    }
+    
+}
+
+extension WeatherDetailViewController: UICollectionViewDelegate {
+    
+    func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
+        collectionView.deselectItem(at: indexPath, animated: true)
+    }
+    
+}
+
+extension WeatherDetailViewController: UICollectionViewDataSource {
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        weatherDetailViewModel.hourlyWeather?.hourlyForecast.count ?? 0
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        let cell = collectionView.dequeueReusableCell(withReuseIdentifier: WeatherCollectionViewCell.identifier, for: indexPath) as! WeatherCollectionViewCell
```

Comment:
Avoid force downcasting (as!). Use optional instead and handle possible failure.